yt_dlp/options.py

```diff
@@ -515,6 +515,18 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
         metavar='IP', dest='source_address', default=None,
         help='Client-side IP address to bind to',
     )
+    network.add_option(
+        '--impersonate',
+        metavar='CLIENT[:[VERSION][:OS[:OS_VERSION]]]', dest='impersonate', default=None,
```

Comment:
version should be separate for each bc it needs to be individually comparable

Reply:
I think we discussed this a while back too. That format would be even more confusing imo

Reply:
> That format would be even more confusing imo

Why?

> version should be separate for each bc it needs to be individually comparable

In the dataclass, yes. I am talking about the CLI only

```py
    def from_str(cls, target):
        return cls(*re.fullmatch(r'(\w+)([\d.]+)?(?::(\w+)([\d.]+)?)?', target).groups())
```

Reply:
I think its preference, and I prefer the old over the new.

Note that this will also affect the `impersonate=` parameter for `download_website` and related.